projects/magnify-cash/index.js

```diff
@@ -0,0 +1,86 @@
+const { getLogs } = require('../helper/cache/getLogs')
+
+const USDC_WORLDCHAIN = "0x79A02482A880bCE3F13e09Da970dC34db4CD24d1"
+const Magnify_Cash_V1 = '0x4E52d9e8d2F70aD1805084BA4fa849dC991E7c88';
+const Magnify_Cash_V2 = '0x93dbB2d447F0086aF60B2becc66598fe3D9135A1';
+
+const tvl = async (api) => {
+  const balanceV1 = await api.call({
+    abi: 'erc20:balanceOf',
+    target: USDC_WORLDCHAIN,
+    params: [Magnify_Cash_V1],
+  });
+
+  const balanceV2 = await api.call({
+    abi: 'erc20:balanceOf',
+    target: USDC_WORLDCHAIN,
+    params: [Magnify_Cash_V2],
+  });
+
+  api.add(USDC_WORLDCHAIN, balanceV1);
+  api.add(USDC_WORLDCHAIN, balanceV2);
+
+}
+const totalBorrowed = async (api) => {
+
+  const logsBorrowV1 = await getLogs({
+    api,
+    target: Magnify_Cash_V1,
+    fromBlock: 8116814,
+    eventAbi: "event LoanRequested (uint256 indexed tokenId, uint256 amount, address borrower)",
+    onlyArgs: true,
+    extraKey: 'borrow-v1'
+  });
+
+  const logsRepaidV1 = await getLogs({
+    api,
+    target: Magnify_Cash_V1,
+    fromBlock: 8141647,
+    eventAbi: "event LoanRepaid (uint256 indexed tokenId, uint256 repaymentAmount, address borrower)",
+    onlyArgs: true,
+    extraKey: 'repay-v1'
+  });
+
+  const logsBorrowV2 = await getLogs({
+    api,
+    target: Magnify_Cash_V2,
+    fromBlock: 10253325,
+    eventAbi: "event LoanRequested (uint256 indexed tokenId, uint256 amount, address borrower)",
+    onlyArgs: true,
+    extraKey: 'borrow-v2'
+  });
+
+  const logsRepaidV2 = await getLogs({
+    api,
+    target: Magnify_Cash_V2,
+    fromBlock: 10253325,
+    eventAbi: "event LoanRepaid (uint256 indexed tokenId, uint256 repaymentAmount, address borrower)",
+    onlyArgs: true,
+    extraKey: 'repay-v2'
+  });
+
+  // Only 2 types of loans 1 USDC and 10 USDC, fixed repayment amount
+  const amountWithoutInterest = (repaymentAmount) => {
+    if (repaymentAmount === BigInt(10150000)) {
+      return BigInt(10000000);
+    } else if (repaymentAmount === BigInt(1025000)) {
+      return BigInt(1000000);
+    } else {
+      return repaymentAmount;
+    }
+  }
+
+  const totalBorrowAmount =  [...logsBorrowV1, ...logsBorrowV2].reduce((sum, log) => sum + log.amount, BigInt(0));
+  const totalRepayAmount = [...logsRepaidV1, ...logsRepaidV2].reduce((sum, log) => sum + amountWithoutInterest(log.repaymentAmount), BigInt(0));
+  const netBorrowed = totalBorrowAmount - totalRepayAmount;
```

Comment:
is there any way to calculate net borrows without using tx logs?

Reply:
Currently there is none.
Would it be better if it calls an API to get net borrows? If so we can build one.